### ISSUE TITLE
FontForge-2017-07-31-Windows-r2 “Expand Stroke”: Internal Error: Some fragments did not join in uni071B

### ISSUE THREAD

**Issue by ilyaza** [fontforge/fontforge#3264]
Instructions: load the attached file, do <kbd>Ctrl</kbd>-<kbd>A</kbd>, <kbd>Ctrl</kbd>-<kbd>Shift</kbd>-<kbd>E</kbd>.  Set `64.3×64.3,   Line Cap=Round,  Remove internal contour`
and start the operation.

All the glyphs in the attached `.sfd` file show the error above.  (This is, actually, an update to obsolete #1424.)

### When reporting a bug/issue:
- [ ] Screenshot
- [x] The FontForge version and the operating system you're using
- [x] The behavior you expect to see, and the actual behavior
- [x] Steps to reproduce the behavior 
- [ ] \(optional) Possible solution/fix/workaround

[filt-unifont-mono-x0.sfd.txt](https://github.com/fontforge/fontforge/files/1799159/filt-unifont-mono-x0.sfd.txt)

**Comment by ilyaza** [comment]
**UPDATE:**:

This is not only a warning.  All these glyphs (and only them) are indeed found by “Find Errors” (<kbd>Ctrl</kbd>-<kbd>E</kbd>) with “Paths/Open paths” checked in.

**Comment by ctrlcctrlv** [comment]
There's nothing _per se_ **wrong** with your test case @ilyaza , but it's not ideal due to how weird the glyphs are and how such (sorry if this offends you) ugly, weird glyphs are unlikely to occur in a real font.

A better test case for intrepid developers hoping to figure out this longstanding issue can be found in glyph `m` inside #3684. It's a much simpler shape yet has the same problem.

**Comment by ctrlcctrlv** [comment]
I found a much simpler test case, doesn't even require Spiro, consists of only corners, and only has four points.

Drum roll please! 

# :drum: :drum: :drum: :drum: 
# [y.sfd](https://github.com/fontforge/fontforge/files/3145351/y.zip)!!

![y](https://user-images.githubusercontent.com/838783/57191769-bfacf100-6f5b-11e9-8054-dc5a3cebe5b1.png)
![y2](https://user-images.githubusercontent.com/838783/57191776-cd627680-6f5b-11e9-81eb-1d0255373734.png)

cc: @skef

**Comment by skef** [comment]
So I've now futzed around with `y.sfd` a little bit, in that I built a little visualizer for `StrokeContext->all left/me/right`. I haven't looked at the curve fitting yet. 

Broadly speaking the way the algorithm works is by stepping through each spline segment slowly (point `me`) and generating `left` and `right` points relative to `me`. All of these points are dumped together into the context array to be fiddled with later. (There are also some annotations on the points to help out with the fitting.) 

This is making me suspicious about how the fitting part works but I'll reserve judgment until I've read through some of it.
